### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by DebyGrey**
## To Do list: add & remove

In this project, I:

- removed all hardcoded items from the tasks array.
- created new JavaScript files for the new functionality.
- implemented a function for adding a new task.
- implemented a function for deleting a task.
- Implemented a function for editing task descriptions.
- implemented the re-order functionality
- implemented the update functionality.
- implemented local storage.

### CODE REVIEW

File: src/modules/UI.js [hunk 1]
```diff
@@ -1,23 +1,204 @@
...
+    if (inputValue === '') {
+      trashBin.disabled = false;
+      trashBin.addEventListener('click', (e) => {
+        Store.removeTask(e, taskID);
+        UI.toDoList = Store.getTasks();
+        UI.reOrderList();
+        UI.displayToDoList();
+      });
+    }
+  }
```
Comment (GedeonTS) [inline src/modules/UI.js]: - Good job implementing all the required funtionalities. However, when I click on the trash button, the task is not being deleted neither from the List, nor from the LocalStorage. Kindly fix that according to [this requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Implement%20a%20function%20for%20deleting%20a%20task%20(remove%20an%20element%20from%20the%20array).)

Screenshot from 2023-05-12 14-58-08

**Review by GedeonTS — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED :recycle: :

Hi @DebyGrey , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :tophat:  :+1: 
- No Linters Errors :100: 
- Good PR Description :1st_place_medal: 
- Good Documentation :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/UI.js (hunk 1 above)
Comment (DebyGrey) [reply src/modules/UI.js]: @GedeonTS  it works actually just that my implementation is different. I thought I saw in the video that the button becomes active only when the user has cleared the input. I will correct it now, thanks.

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @DebyGrey, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**